spring-boot-project/spring-boot-properties-migrator/src/main/java/org/springframework/boot/context/properties/migrator/PropertiesMigrationReporter.java

```diff
@@ -105,10 +107,53 @@ PropertiesMigrationReport getReport() {
 				result.add(name,
 						new PropertyMigration(configurationProperty, metadata, determineReplacementMetadata(metadata)));
 			}
+			getMatchingPropertiesForMapEntries(name, source, metadata, result);
 		}));
 		return result;
 	}
 
+	private void getMatchingPropertiesForMapEntries(String name, ConfigurationPropertySource source,
+			ConfigurationMetadataProperty metadata, MultiValueMap<String, PropertyMigration> result) {
+		if (source.getUnderlyingSource() instanceof MapPropertySource && name.equals(metadata.getName())) {
```

Comment:
I don't understand what this is doing. `MapPropertySource` is an implementation detail and you shouldn't be looking at it. It has nothing to do with a deprecated property that is of type `Map`.

Reply:
I see, I thought MapPropertySource indicated the Map deprecation. But if we don't have a way to filter for Map deprecations than agreed it must be done by prefix matching on the name itself.